schedview/app/scheduler_dashboard/scheduler_dashboard.py

```diff
@@ -1274,6 +1285,32 @@ def map_title(self):
         return self.map_title_pane
 
 
+class RestrictedFilesScheduler(Scheduler):
+    """A Parametrized container for parameters, data, and panel objects for the
+    scheduler dashboard with restricted pickle file selection.
+    """
+
+    # Param parameters that are modifiable by user actions.
+    scheduler_fname_doc = """URL or file name of the scheduler pickle file.
+    Such a pickle file can either be of an instance of a subclass of
+    rubin_sim.scheduler.schedulers.CoreScheduler, or a tuple of the form
+    (scheduler, conditions), where scheduler is an instance of a subclass of
+    rubin_sim.scheduler.schedulers.CoreScheduler, and conditions is an
+    instance of rubin_sim.scheduler.conditions.Conditions.
+    """
+    scheduler_fname = schedview.param.FileSelectorWithEmptyOption(
+        doc=scheduler_fname_doc,
+        default="",
+        allow_None=True,
+    )
+
+    def __init__(self, data_dir=None):
+        super().__init__()
+
+        if data_dir is not None:
+            self.param.scheduler_fname.path = f"{data_dir}/*scheduler*.p*"
```

Comment:
I think you should be able to fix the problem of ignoring the data_dir as set by the command line argument by replacing this assignment with something like:
`self.param['scheduler_fname'].update(path=f"{data_dir}/*scheduler*.p*")`

Reply:
Thank you. That worked